src/main/java/org/jabref/gui/frame/JabRefFrame.java

```diff
@@ -223,12 +244,42 @@ private void initLayout() {
         head.setSpacing(0d);
         setTop(head);
 
-        splitPane.getItems().addAll(tabbedPane);
+        contentPane.getChildren().addAll(welcomePage, tabbedPane);
+        splitPane.getItems().add(contentPane);
+
         SplitPane.setResizableWithParent(sidePane, false);
         sidePane.widthProperty().addListener(c -> updateSidePane());
         sidePane.getChildren().addListener((InvalidationListener) c -> updateSidePane());
-        updateSidePane();
+
         setCenter(splitPane);
+        updateContent();  // Ensure layout is consistent on load
+    }
+
+    private void updateContent() {
+        boolean hasOpenDatabases = !stateManager.getOpenDatabases().isEmpty();
+
+        // Toggle visibility between welcome page and tabbed pane
+        welcomePage.setVisible(!hasOpenDatabases);
+        tabbedPane.setVisible(hasOpenDatabases);
+
+        if (hasOpenDatabases) {
+            if (!splitPane.getItems().contains(sidePane)) {
+                sidePane.setPrefWidth(260);
+                splitPane.getItems().addFirst(sidePane);
+
+                // Ensure divider position is set before rendering
+                splitPane.setDividerPositions(0.2);
+
+                // Ensure smooth transition by updating the layout immediately
+                Platform.runLater(() -> splitPane.setDividerPositions(0.2));
```

Comment:
I don't understand that. Both values are equal. I think, only the statement on line 271 is required?

The code of `org.jabref.gui.frame.JabRefFrame#updateDividerPosition` shows different values. I think you should just call

    splitPane.setDividerPositions(preferences.getGuiPreferences().getSidePaneWidth() / splitPane.getWidth());

Reply:
You are right - `splitPane.setDividerPositions(preferences.getGuiPreferences().getSidePaneWidth() / splitPane.getWidth());`  works fine!

https://github.com/JabRef/jabref/pull/12017/commits/eab1844ab1e92300be98cb1090148d2e6bc0a6bc